packages/composite-checkout-wpcom/src/hooks/use-shopping-cart.ts

```diff
@@ -42,6 +42,7 @@ type ShoppingCartHookState = {
 	responseCart: ResponseCart;
 	couponStatus: CouponStatus;
 	cacheStatus: CacheStatus;
+	location: CartLocation;
```

Comment:
Since we already have a location object in responseCart, what do you think about using it instead of adding this to the state?

I think the action could look something like

```
case 'SET_LOCATION':
  return {
    ...state,
    responseCart: setTaxLocationInResponseCart( state.responseCart, action.location );
  }
```

where `setTaxLocationInResponseCart` would do the actual cart edit and be defined in `types/backend/shopping-cart-endpoint`, analogous to remove/add item and add coupon.

Reply:
Yup. I should have started with that, but as you saw, I got there eventually. 😁